File: social-service/src/main/java/com/eirsteir/coffeewithme/social/service/friendship/FriendshipService.java
package com.eirsteir.coffeewithme.social.service.friendship;

import com.eirsteir.coffeewithme.commons.dto.UserDetailsDto;
import com.eirsteir.coffeewithme.social.domain.friendship.FriendshipId;
import com.eirsteir.coffeewithme.social.domain.friendship.FriendshipStatus;
import com.eirsteir.coffeewithme.social.domain.user.User;
import com.eirsteir.coffeewithme.social.dto.FriendshipDto;
import com.eirsteir.coffeewithme.social.web.request.FriendRequest;
import java.util.List;

public interface FriendshipService {

  List<FriendshipDto> findFriendshipsOf(Long id, FriendshipStatus status);

  List<FriendshipDto> findFriendshipsAtUniversity(User user);

  FriendshipDto registerFriendship(FriendRequest friendRequest);

  boolean friendshipExists(FriendshipId friendshipId);

  void removeFriendship(FriendshipDto friendshipDto);

  FriendshipDto updateFriendship(FriendshipDto friendshipDto);

  List<FriendshipDto> findFriendshipsOf(UserDetailsDto user);

  List<FriendshipDto> findAllFriendshipsWithStatus(UserDetailsDto user, FriendshipStatus status);

  Integer getFriendsCount(Long userId);
}

File: social-service/src/main/kotlin/com/eirsteir/coffeewithme/social/service/friendship/FriendshipService.kt
package com.eirsteir.coffeewithme.social.service.friendship

import com.eirsteir.coffeewithme.commons.dto.UserDetailsDto
import com.eirsteir.coffeewithme.social.domain.friendship.FriendshipId
import com.eirsteir.coffeewithme.social.domain.friendship.FriendshipStatus
import com.eirsteir.coffeewithme.social.domain.user.User
import com.eirsteir.coffeewithme.social.dto.FriendshipDto
import com.eirsteir.coffeewithme.social.web.request.FriendRequest

interface FriendshipService {

    fun findFriendshipsOf(id: Long, status: FriendshipStatus): List<FriendshipDto>

    fun findFriendshipsAtUniversity(user: User): List<FriendshipDto>

    fun registerFriendship(friendRequest: FriendRequest): FriendshipDto

    fun friendshipExists(friendshipId: FriendshipId): Boolean

    fun removeFriendship(friendshipDto: FriendshipDto)

    fun updateFriendship(friendshipDto: FriendshipDto): FriendshipDto

    fun findFriendshipsOf(user: UserDetailsDto): List<FriendshipDto>

    fun findAllFriendshipsWithStatus(
        user: UserDetailsDto,
        status: FriendshipStatus
    ): List<FriendshipDto>

    fun getFriendsCount(userId: Long): Int
}